### PR TITLE
Add image size checkmarks to context menu

### PR DESCRIPTION
**Pull request by ianeli1**
Adds checkmarks to the "Size" menu inside context menu

image

For small, medium, and original sizes, `isResizedTo` is used to check the sizing. This function has been modified to allow for resized images with odd sizes, such as 271px, as this would get divided in half and then rounded up, while HTML would have the number divided in half and rounded down, which would cause issues.

For best fit, we check if "width" and "height" attributes are set.

### CODE REVIEW

File: packages-ui/roosterjs-react/lib/contextMenu/menus/createImageEditMenuProvider.tsx [hunk 1]
```diff
@@ -188,6 +203,14 @@ function shouldShowImageEditItems(editor: IEditor, node: Node) {
     return safeInstanceOf(node, 'HTMLImageElement') && node.isContentEditable;
 }
 
+function getImageFromTarget(target: Node) {
```
Comment (juliaroldi) [inline packages-ui/roosterjs-react/lib/contextMenu/menus/createImageEditMenuProvider.tsx:206]: which scenario triggers this code? Maybe we should consider add it to image selection code and relay only in 'getSelectionRangeEx'

Comment (JiuqingSong) [reply packages-ui/roosterjs-react/lib/contextMenu/menus/createImageEditMenuProvider.tsx:206]: For roosterjs-react, I think it is ok to only support image selection.

File: packages-ui/roosterjs-react/lib/contextMenu/types/ContextMenuItem.ts [hunk 2]
```diff
@@ -42,6 +42,8 @@ export default interface ContextMenuItem<TString extends string, TContext = neve
      */
     shouldShow?: (editor: IEditor, targetNode: Node, context?: TContext) => boolean;
 
+    getSelectedId?: (editor: IEditor, targetNode: Node) => TString | null;
```
Comment (JiuqingSong) [inline packages-ui/roosterjs-react/lib/contextMenu/types/ContextMenuItem.ts:45]: Please add comment

File: packages-ui/roosterjs-react/lib/contextMenu/menus/createImageEditMenuProvider.tsx [hunk 3]
```diff
@@ -77,6 +78,20 @@ const ImageResizeMenuItem: ContextMenuItem<ImageEditMenuItemStringKey, ImageEdit
             }
         });
     },
+    getSelectedId: (editor, target) => {
```
Comment (juliaroldi) [inline packages-ui/roosterjs-react/lib/contextMenu/menus/createImageEditMenuProvider.tsx:81]: I see you are passing the node parameter for this function in the interface. Should we use the node her instead getting the selection?

File: packages/roosterjs-editor-core/lib/corePlugins/ImageSelection.ts [hunk 4]
```diff
@@ -54,7 +59,8 @@ export default class ImageSelection implements EditorPlugin {
                     if (
                         mouseSelection &&
                         mouseSelection.type === SelectionRangeTypes.ImageSelection &&
-                        mouseSelection.image !== mouseTarget
+                        (!safeInstanceOf(mouseTarget, 'HTMLElement') ||
+                            !mouseTarget.contains(mouseSelection.image))
```
Comment (juliaroldi) [inline packages/roosterjs-editor-core/lib/corePlugins/ImageSelection.ts:63]: What happens if the mouseTarget has more content than just image the image?

Comment (ianeli1) [reply packages/roosterjs-editor-core/lib/corePlugins/ImageSelection.ts]: reverted

File: packages/roosterjs-editor-core/lib/corePlugins/ImageSelection.ts [hunk 5]
```diff
@@ -43,7 +43,7 @@ export default class ImageSelection implements EditorPlugin {
                     if (
                         safeInstanceOf(target, 'HTMLImageElement') &&
                         target.isContentEditable &&
-                        event.rawEvent.button === mouseLeftButton
+                        event.rawEvent.button != mouseMiddleButton
```
Comment (juliaroldi) [inline packages/roosterjs-editor-core/lib/corePlugins/ImageSelection.ts:46]: do we need this change?

File: packages/roosterjs-editor-core/lib/corePlugins/DOMEventPlugin.ts [hunk 6]
```diff
@@ -233,11 +233,15 @@ export default class DOMEventPlugin implements PluginWithState<DOMEventPluginSta
...
         const searcher = this.editor?.getContentSearcherOfCursor();
         const elementBeforeCursor = searcher?.getInlineElementBefore();
+        const selection = this.editor?.getSelectionRangeEx();
 
         let eventTargetNode = event.target as Node;
         if (event.button != 2 && elementBeforeCursor) {
             eventTargetNode = elementBeforeCursor.getContainerNode();
         }
+        if (event.button == 2 && selection?.type === SelectionRangeTypes.ImageSelection) {
+            eventTargetNode = selection.image;
```
Comment (juliaroldi) [inline packages/roosterjs-editor-core/lib/corePlugins/DOMEventPlugin.ts:243]: Do we need this changes?

File: packages/roosterjs-editor-core/lib/corePlugins/ImageSelection.ts (hunk 5 above)
Comment (ianeli1) [reply packages/roosterjs-editor-core/lib/corePlugins/ImageSelection.ts:46]: ImageSelection is broken on context menu without it

File: packages/roosterjs-editor-core/lib/corePlugins/DOMEventPlugin.ts (hunk 6 above)
Comment (ianeli1) [reply packages/roosterjs-editor-core/lib/corePlugins/DOMEventPlugin.ts]: Not really, but having image selection be the default provider of the node in this case makes sense. We rarely use the wrapper target on any action

File: packages/roosterjs-editor-core/lib/corePlugins/ImageSelection.ts (hunk 5 above)
Comment (JiuqingSong) [inline packages/roosterjs-editor-core/lib/corePlugins/ImageSelection.ts:46]: There is a copy of this file under content model folder: https://github.com/microsoft/roosterjs/blob/master/packages-content-model/roosterjs-content-model-adapter/lib/corePlugins/ImageSelection.ts

Would you please do the same change to the copied file? Thanks.

Comment (ianeli1) [reply packages/roosterjs-editor-core/lib/corePlugins/ImageSelection.ts:46]: done

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**